src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2350,8 +2356,28 @@ because xbuild doesn't support framework reference assemblies.
 	Condition="Exists('$(_CustomViewMapFile)')"
 	CustomViewMapFile="$(_CustomViewMapFile)"
 	AcwMapFile="$(_AcwMapFile)"
-	ResourceDirectories="$(MonoAndroidResDirIntermediate);@(LibraryResourceDirectories)"
+	ResourceDirectories="$(MonoAndroidResDirIntermediate);@(_LibraryResourceHashDirectories)"
 	ResourceNameCaseMap="$(_AndroidResourceNameCaseMap)"
+  >
+    <Output TaskParameter="Processed" ItemName="_ProcessedCustomViews" />	
+  </ConvertCustomView>
+  <Delete Files="@(_ProcessedCustomViews->'$(_AndroidLibraryFlatArchivesDirectory)%(Hash).stamp')"
+    Condition=" '$(AndroidUseAapt2)' == 'True' And '@(_ProcessedCustomViews)' != '' "
+  />
+  <Aapt2Compile
+    Condition=" '$(AndroidUseAapt2)' == 'True' And '@(_ProcessedCustomViews)' != '' "
+    ContinueOnError="$(DesignTimeBuild)"
+    ResourceDirectories="@(_LibraryResourceHashDirectories);$(MonoAndroidResDirIntermediate)"
+    ExplicitCrunch="$(AndroidExplicitCrunch)"
+    ExtraArgs="$(AndroidAapt2CompileExtraArgs)"
+    FlatArchivesDirectory="$(_AndroidLibraryFlatArchivesDirectory)"
+    ToolPath="$(Aapt2ToolPath)"
+    ToolExe="$(Aapt2ToolExe)">
+    <Output TaskParameter="CompiledResourceFlatArchives" ItemName="_UpdatedFlatArchives" />	
+  </Aapt2Compile>
+  <Touch Files="@(_UpdatedFlatArchives->'$(_AndroidLibraryFlatArchivesDirectory)\%(Filename).stamp')"
+    Condition=" '$(AndroidUseAapt2)' == 'True' And '@(_UpdatedFlatArchives)' != '' "
+    AlwaysCreate="True"
```

Comment:
This is the spot, I was wondering if the test does a build with no changes. Wanting to make sure this `<Touch/>` is doing what we want.